packages/puppeteer-core/src/api/locators/locators.ts

```diff
@@ -113,6 +114,18 @@ export abstract class Locator<T> extends EventEmitter<LocatorEvents> {
     return RaceLocator.create(locators);
   }
 
+  /**
+   * Trying to locate all the elements for locator array in parallel
+   * and return the last locator that locate element.
+   *
+   * @public
+   */
+  static last<Locators extends readonly unknown[] | []>(
```

Comment:
could you elaborate more about the testing scenario you have?  Can you use the following instead?

```
await Promise.all([
  page.locator('#1').wait(),
  page.locator('#2').wait()
]);
await page.locator('#3').click();
```

It is not as parallelized but it does not have the following problem:

```

const result = Locator.last([
    page.locator('#test1'),
    page.locator('#test2'),
    page.locator('#test3'),
  ])
    .setTimeout(5000)
    .click();
```

The last locator never intends to click on the first two located elements but the preconditions from the click action would also apply when locating those elements which does not make sense since they are never clicked.

Reply:
Hi @OrKoN ,
The testing scenario is like you suggested. The third locater depends on the first two locators. The third locater can do some operations like click only after the appearances of the first two elements.
The code is like this:
await Locator.last([locator1, locator2, locator3]).wait();
await locator3.click();
I think it is more convenient to provide such a method and it may be faster since it is paralleled.

Reply:
so `wait` and `click` add different preconditions, so if you use Locator.last with `click` you would get different results compared to using it with `wait`. I am concerned that it might be unexpected for the users, who would expect the wait conditions to apply to the first locator, and the click conditions apply only to the last locator.

Reply:
actually, I might be misremembering the locator implementation. @Lightning00Blade would you mind giving a second opinion?